### PR TITLE
Enable peer tags aggregation by default (REPEATS PR #30232)

### PR DESCRIPTION
**Pull request by rickbatka**
### Repeating [previously reverted PR](https://github.com/DataDog/datadog-agent/pull/30232).
This PR repeats [this PR](https://github.com/DataDog/datadog-agent/pull/30232), which was reverted because it caused some end-to-end test failures.

It includes an additional commit that fixes the test in question.

============================================================
### Previous PR notes below:
============================================================

https://datadoghq.atlassian.net/browse/APMST-1371

What does this PR do?
Flips default setting to "true" for three flags:

apm_config.peer_service_aggregation
apm_config.peer_tags_aggregation
apm_config.compute_stats_by_span_kind
Motivation
The intent is to enable peer tags aggregation (and the related "stats by span kind" aggregation) as the default option, effectively moving from "Opt-In" to "Opt-Out" for the Service Representation project (docs [here](https://docs.datadoghq.com/tracing/guide/inferred-service-opt-in/?tab=java)). This project is moving toward General Availability and we would like to land this before the Nov 1 code freeze.

This is a non-breaking change: all back-end data and pipelines are already updated, and inferred entities are appearing throughout APM. This setting ensures that those inferred entities will have stats properly aggregated and attributed to them.

Describe how to test/QA your changes
I am unsure how to thoroughly test the trace agent. I have updated and added unit tests, but I would greatly appreciate some help confirming the changes or devising a test plan to help confirm the changes!

Possible Drawbacks / Trade-offs
This shouldn't have any breaking implications for our customers.

However, as the comments in the template config suggest, the additional aggregation could cause resource usage to increase. We have a few hundred customers using the feature today, but it is possible a wide rollout could expose more performance issues. I'm also looking for guidance from the team on if / how we need to address any such performance regression testing.

### CODE REVIEW

**Comment by rickbatka** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by rickbatka** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

File: test/new-e2e/tests/otel/utils/pipelines_utils.go [hunk 1]
```diff
@@ -31,13 +31,14 @@ import (
...
-	calendarService      = "calendar-rest-go"
-	telemetrygenService  = "telemetrygen-job"
-	env                  = "e2e"
-	version              = "1.0"
-	customAttribute      = "custom.attribute"
-	customAttributeValue = "true"
-	logBody              = "random date"
+	calendarService              = "calendar-rest-go"
+	telemetrygenService          = "telemetrygen-job"
+	telemetrygenTopLevelResource = "lets-go"
```
Comment (ichinaski) [inline test/new-e2e/tests/otel/utils/pipelines_utils.go:36]: where is this value coming from?

**Review by ichinaski — APPROVED** [review]

**Review by maycmlee — APPROVED** [review]

**Review by liustanley — APPROVED** [review]

**Comment by rickbatka** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Review by pgimalac — APPROVED** [review]

**Pull request merged**